src/modules/core/components/Fields/InlineEdit/MultiLineEdit/MultiLineEdit.tsx

```diff
@@ -92,112 +89,87 @@ interface Props {
-    if (!allowReturns) {
-      return HANDLED;
-    }
-    return NOT_HANDLED;
-  };
-
-  onBlur = (evt: FocusEvent<EditorType>) => {
-    const { onEditorBlur, setValue, $value } = this.props;
-    let editorState = $value;
-    if (!this.editorHasContent()) {
-      editorState = EditorState.push(
-        editorState,
-        ContentState.createFromText(''),
-        'remove-range',
-      );
-      setValue(editorState);
-    }
-    if (onEditorBlur) {
-      onEditorBlur(evt, editorState);
-    }
-  };
-
-  onChange = (editorState: EditorStateType) => {
-    const { setValue } = this.props;
-    setValue(editorState);
-  };
-
-  render() {
-    const {
-      appearance,
-      elementOnly,
-      $error,
-      extra,
-      formatIntl,
-      help,
-      helpValues,
-      $id,
-      label,
-      labelValues,
-      name,
-      placeholder,
-      placeholderValues,
-      readOnly,
-      spellCheck,
-      status,
-      statusValues,
-      $value,
-    } = this.props;
-    const placeholderText =
-      typeof placeholder === 'object'
-        ? formatIntl(placeholder, placeholderValues)
-        : placeholder;
-    return (
-      <div className={getMainClasses(appearance, styles)}>
-        {!elementOnly && (
-          <InputLabel
-            error={$error}
-            help={help}
-            helpValues={helpValues}
-            extra={extra}
-            label={label}
-            labelValues={labelValues}
-          />
-        )}
-        <Editor
-          name={name}
-          editorKey={$id}
-          editorState={$value}
-          handleReturn={this.handleReturn}
-          onBlur={this.onBlur}
-          onChange={this.onChange}
-          placeholder={placeholderText}
-          readOnly={readOnly}
-          spellCheck={spellCheck}
-          stripPastedStyles
-        />
-        <InputStatus
-          status={status}
-          statusValues={statusValues}
+const createEditorState = ($value: string) =>
+  $value
+    ? EditorState.createWithContent(ContentState.createFromText($value))
+    : EditorState.createEmpty();
+
+const MultiLineEdit = ({
+  $error,
+  $id,
+  $value,
+  allowReturns = true,
+  appearance,
+  extra,
+  formatIntl,
+  help,
+  helpValues,
+  label,
+  labelValues,
+  name,
+  onEditorBlur,
+  placeholder,
+  placeholderValues,
+  setValue,
+  status,
+  statusValues,
+  elementOnly = true,
+  readOnly = false,
+  spellCheck = false,
+}: Props) => {
+  const [editorState, setEditorState] = useState<EditorStateType>(() =>
+    createEditorState($value),
+  );
+  useEffect(() => setEditorState(createEditorState($value)), [$value]);
+  const handleReturn = useCallback(
+    () => (!allowReturns ? HANDLED : NOT_HANDLED),
+    [allowReturns],
+  );
+  const onBlur = useCallback(
+    (evt: FocusEvent) => {
+      if (onEditorBlur) {
+        const content = editorState.getCurrentContent();
+        const newValue =
+          (content.hasText() && content.getPlainText().trim()) || '';
```

Comment:
I actually thought the `.trim()` would take care of the spaces issue. I'll look into it more